kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/metrics/SnapshotReportSuite.scala

```diff
@@ -463,24 +520,26 @@ class SnapshotReportSuite extends AnyFunSuite with MetricsReportTestUtils {
         // No CRC for golden table
         expectNonZeroDurationToGetCrcInfo = false))
 
-    // Test getSnapshotAsOfTimestamp
-    // We use the timestamp of version 0
-    val version0Timestamp = new File(FileNames.deltaFile(new Path(path, "_delta_log"), 0))
-      .lastModified()
-    checkSnapshotReport(
-      (table, engine) => table.getSnapshotAsOfTimestamp(engine, version0Timestamp),
-      path,
-      SnapshotReportExpectations(
-        expectedReportCount = 1,
-        expectException = true,
-        // Query will fail before timestamp -> version resolution. The failure
-        // will happen when `getLatestSnapshot` is called.
-        expectedVersion = Optional.of(0),
-        expectedCheckpointVersion = Optional.empty(),
-        expectedProvidedTimestamp = Optional.empty(),
-        // This is due to the `getLatestSnapshot` call
-        expectNonZeroLoadProtocolAndMetadataDuration = true,
-        // No CRC for golden table
-        expectNonZeroDurationToGetCrcInfo = false))
+    // Test getSnapshotAsOfTimestamp - only if adapter supports it
+    if (tableManager.supportsTimestampResolution) {
```

Comment:
all we added was this if statement, and then update to use the table manager adapter